controllers/imageupdateautomation_controller.go

```diff
@@ -358,21 +366,28 @@ func (r *ImageUpdateAutomationReconciler) Reconcile(ctx context.Context, req ctr
 		return failWithError(err)
 	}
 
-	// The status message depends on what happens next. Since there's
-	// more than one way to succeed, there's some if..else below, and
-	// early returns only on failure.
-	if rev, err := gitClient.Commit(
-		git.Commit{
-			Author: git.Signature{
-				Name:  gitSpec.Commit.Author.Name,
-				Email: gitSpec.Commit.Author.Email,
-				When:  time.Now(),
+	var rev string
+	if len(templateValues.Updated.Files) > 0 {
+		// The status message depends on what happens next. Since there's
+		// more than one way to succeed, there's some if..else below, and
+		// early returns only on failure.
+		rev, err = gitClient.Commit(
+			git.Commit{
+				Author: git.Signature{
+					Name:  gitSpec.Commit.Author.Name,
+					Email: gitSpec.Commit.Author.Email,
+					When:  time.Now(),
+				},
+				Message: message,
 			},
-			Message: message,
-		},
-		repository.WithSigner(signingEntity),
-	); err != nil {
-		if err != git.ErrNoStagedFiles {
+			repository.WithSigner(signingEntity),
+		)
+	} else {
+		err = extgogit.ErrEmptyCommit
+	}
```

Comment:
why are we creating an error here?

Reply:
This is to be able to reuse the existing logic for handling empty commits.

Given that this controller will be refactored as part of the GA process, I found this was the easiest way to resolve the problem without duplicating code.